madlibs.js

```diff
@@ -1,42 +1,108 @@
- * Output: [
- *  { word: "Louis", pos: "noun" },
- *  { word: "went", pos: "verb", },
- *  { word: "to", },
- *  { word: "the", },
- *  { word: "store", pos: "noun" }
- *  { word: "," }
- *  ....
- * 
- * There are multiple ways to do this, but you may want to use regular expressions.
- * Please go through this lesson: https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/regular-expressions/
- */
+let objectsArray = [];
+//This function will be used for alert 
+function onlyLetters(str) {
+  return /^[A-Za-z]*$/.test(str);
+}
 function parseStory(rawStory) {
-  // Your code here.
-  return {}; // This line is currently wrong :)
+  let storyArr = rawStory.split(" ");
+  storyArr.forEach((element) => {
+    let storyWord = element.match(/\w+(?=\s*\[)/g);
+    let wordType = element.substr(-3);
+    let checkWordType = () => {
+      if (wordType === "[a]") {
+        return "adjective";
+      } else if (wordType === "[n]") {
+        return "noun";
+      } else if (wordType === "[v]") {
+        return "verb";
+      }
+    };
+    if (storyWord) {
+      return objectsArray.push({ word: element.slice(0, -3), pos: checkWordType() });
+    } 
+    else {
+      return objectsArray.push({ word: element });
+    }
+  });
+  let inputValue;
+  let p = document.getElementById("paragraph");
+  objectsArray.map((object, index) => {
+    if (object.pos) {
+      let input = document.createElement("input");
+
+      input.setAttribute("type", "text");
+      input.setAttribute("placeholder", object.pos);
+      input.setAttribute("class", "inputs");
+      input.setAttribute("maxLength", "20");
+      p.appendChild(input);
+      input.addEventListener("keydown", (e) => {
+        if (!onlyLetters(e.key)) {
+          e.preventDefault();
+          alert("Please enter alphabetic characters");
+        }
+      })
+
+      input.addEventListener("input", (e) => {
+        inputValue = e.target.value;
+        let x = document.getElementById(index);
+        x.innerText = e.target.value;
+      });
+    } else {
+      let text = document.createTextNode(" " + object.word);
+      p.appendChild(text);
+      return;
+    }
+  });
+//Preview mode
+  let p2 = document.getElementById("paragraph2");
+  objectsArray.map((object, index) => {
+    if (object.pos) {
+      valueSpan = document.createElement("span");
+      valueSpan.setAttribute("id", index);
+      p2.appendChild(valueSpan);
+    } else {
+      text = document.createTextNode(" " + object.word + " ");
+      p2.appendChild(text);
+    }
+  });
+  let inputs = document.getElementsByClassName("inputs");
+  for (let i = 0; i < inputs.length; i++) {
+    inputs[i].addEventListener("keydown", (e) => {
+      if (e.which === 13 || e.which === 39) {
+        e.preventDefault();
+        inputs[i].nextElementSibling.focus();
+      } else if (e.which === 37) {
+        e.preventDefault();
+        inputs[i].previousElementSibling.focus();
+      }
+    });
+  }
```

Comment:
```suggestion
  inputs.forEach(input => {
     input.addEventListener("keydown", (e) => {
      e.preventDefault();
      if (e.which === 13 || e.which === 39) {
        inputs[i].nextElementSibling.focus();
      } else if (e.which === 37) {
        inputs[i].previousElementSibling.focus();
      }
    });
  });

```
Amazing way of doing the keydown event. Just small adjustments to make it awesome.